VMEncryption/main/DiskUtil.py

```diff
@@ -957,7 +956,24 @@ def is_os_disk_lvm(self):
 
         return DiskUtil.os_disk_lvm
 
-    def should_skip_for_inplace_encryption(self, device_item, encrypt_volume_type):
+    def is_data_disk(self, device_item, azure_devices):
+        # Root disk
+        if device_item.device_id.startswith('00000000-0000'):
+            self.logger.log(msg="skipping root disk", level=CommonVariables.WarningLevel)
+            return False
+        # Resource Disk. Not considered a "data disk" exactly (is not attached via portal and we have a separate code path for encrypting it)
+        if device_item.device_id.startswith('00000000-0001'):
+            self.logger.log(msg="skipping resource disk", level=CommonVariables.WarningLevel)
+            return False
+
+        for azure_blk_item in azure_devices:
```

Comment:
same as single pass -- would like to see a comment making it clear the first checks are for gen 1 images and the second are for gen 2 images, also if there is any chance of additional devices beyond root and resource disk being provided since it is being passed as a parameter.   final case log message could be updated to reflect the exact device it is skipping by name in the log message instead of ambiguous "azure disk root or resource"